[Python] Drop chip-device-ctrl (#33488)
project-chip/connectedhomeip: docs/guides/python_chip_controller_building.md

Reviewer: nit, if we do plan on keeping this file around with it's content, should we rename the file to reflect the content a little more. It seems like it has expanded greatly from building the python chip controller

Author: The current name doesn't seem too far of to me, no?

There are two things: The Python CHIP Controller (library), which is essentially almost all the Python code which allows to be a CHIP Controller using Python. And then there is a (rather small) REPL built on-top of it. While the docs document mostly the API as it can be used from the REPL, I think just documenting how-to build the Python stuff makes sense.

### Step 6: Control application clusters.

For the light bulb example, execute the following command to toggle the LED
state:

```
await devCtrl.SendCommand(1234, 1, Clusters.OnOff.Commands.Toggle())
```

To change the brightness of the LED, use the following command, with the level
value somewhere between 0 and 255.

```
commandToSend = LevelControl.Commands.MoveToLevel(level=50, transitionTime=Null, optionsMask=0, optionsOverride=0)
await devCtrl.SendCommand(1234, 1, commandToSend)
```

### Step 7: Read basic information out of the accessory.

Every Matter accessory device supports a Basic Information Cluster, which
maintains collection of attributes that a controller can obtain from a device,
such as the vendor name, the product name, or software version. Use
`ReadAttribute()` command to read those values from the device:

```
attributes = [
    (0, Clusters.BasicInformation.Attributes.VendorName),
    (0, Clusters.BasicInformation.Attributes.ProductName),
    (0, Clusters.BasicInformation.Attributes.SoftwareVersion),
]
await devCtrl.ReadAttribute(1234, attributes)
```

> In REPL, you can type `Clusters.BasicInformation.Attributes.` and then use the
> TAB key.

<hr>

> ## List of commands

Reviewer 2: We now have a full API description from the ChipDeviceCtrl documentation here (https://project-chip.github.io/connectedhomeip-doc/testing/ChipDeviceCtrlAPI.html) - it's generated from the function comments.

I think this documentation was written before some of the functions were updated, and before some of the new commissioning stuff landed, so there are a few things in here that are a bit out of date. WDYT about just getting rid of this section and making a note that the devCtrl is a device controller and you can use any of the normal functions, and then just linking over to that documentation?

### `SetThreadOperationalDataset(<thread-dataset>)`

Provides the controller with Thread network credentials that will be used in the
device commissioning procedure to configure the device with a Thread interface.

```
thread_dataset = bytes.fromhex("0e080000000000010000000300001335060004001fffe002084fe76e9a8b5edaf50708fde46f999f0698e20510d47f5027a414ffeebaefa92285cc84fa030f4f70656e5468726561642d653439630102e49c0410b92f8c7fbb4f9f3e08492ee3915fbd2f0c0402a0fff8")
devCtrl.SetThreadOperationalDataset(thread_dataset)
```

### `SetWiFiCredentials(<ssid>: str, <password>: str)`

Provides the controller with Wi-Fi network credentials that will be used in the
device commissioning procedure to configure the device with a Wi-Fi interface.

```
devCtrl.SetWiFiCredentials('TESTSSID', 'P455W4RD')
```

> ### `CommissionIP(<ip address>: str, <setup pin code>: int, <nodeid>: int)`

Reviewer 2: This function is marked as deprecated in the API.

Reviewer: +1, this is replaced with CommissionOnNetwork.

Author: Ok, since it is currently marked deprecated, we should drop it from here. But I do disagree on the deprecation of this API (see discussion in #tsg-tt-python-repl). But that is a different discussion.

> To be implemented in REPL

Print the generated Onboarding Payload Contents in human-readable (Manual
Pairing Code) and machine-readable (QR Code) format:

### `setup-payload parse-manual <manual-pairing-code>`

> To be implemented in REPL

Print the commissioning information encoded in the Manual Pairing Code:

### `setup-payload parse-qr <qr-code>`

> To be implemented in REPL

Print the commissioning information encoded in the QR Code payload:

### `devCtrl.SendCommand(<nodeid>: int, <endpoint>: int, Clusters.<cluster>.Commands.<command>(<arguments>))`

Send a Matter command to the device. For example:

```python
commandToSend = Clusters.LevelControl.Commands.MoveWithOnOff(moveMode=1, rate=2, optionsMask=0, optionsOverride=0)
await devCtrl.SendCommand(1234, 1, commandToSend)
```

To see available arguments just create a command object without argument:

```
Clusters.LevelControl.Commands.MoveWithOnOff()
```

Shows which arguments are available:

```
MoveWithOnOff(
│   moveMode=0,
│   rate=Null,
│   optionsMask=0,
│   optionsOverride=0
)
```

### `ReadAttribute(<nodeid>: int, [(<endpoint id>: int, Clusters.<cluster>.Attributes.<attribute>)])`

> Read the value of ZCL attribute. For example:

Reviewer 2: ```suggestion
Read the value of an attribute. For example:
```

```python
await devCtrl.ReadAttribute(1234, [(0, Clusters.BasicInformation.Attributes.VendorName)])
```

### `WriteAttribute(<nodeid>: int, [(<endpointid>: int, Clusters.<cluster>.Attributes.<attribute>(value=<attribute value>))])`

> Write the value to a ZCL attribute. For example:

Reviewer 2: ```suggestion
Write a value to an attribute. For example:
```
